bundles/model/org.eclipse.smarthome.model.sitemap/src/org/eclipse/smarthome/model/validation/SitemapValidator.xtend

```diff
@@ -83,4 +85,34 @@ class SitemapValidator extends AbstractSitemapValidator {
 		}
 	}
 
+	@Check
+	def void checkSetpoints(Setpoint sp) {
+	    if(sp.item === null)
+	    {
+	        error("Setpoint without an item defined",
+                    SitemapPackage.Literals.SETPOINT.getEStructuralFeature(SitemapPackage.SETPOINT));
+            return
+	    }
+
+	    if(BigDecimal.ZERO.equals(sp.step))
+	    {
+	        error("Setpoint on item '" +sp.item+ "' has step size of 0",
+                    SitemapPackage.Literals.SETPOINT.getEStructuralFeature(SitemapPackage.SETPOINT__STEP));
+	        return
+	    }
+
+	    if(sp.step < BigDecimal.ZERO)
```

Comment:
Why can you use `<` here but had to use `equals` above? If Xtend can handle this, then you probably also can use `0` directly, have you tried that?

Reply:
Indeed I can, I tried that before by it did not work because it was friday and I was confusing `BigInteger` with `BigDecimal` :P